src/pybind/mgr/dashboard/services/auth.py

```diff
@@ -277,3 +288,64 @@ def _check_authorization(self, username):
         if not AuthManager.authorize(username, sec_scope, sec_perms):
             raise cherrypy.HTTPError(403, "You don't have permissions to "
                                           "access that resource")
+
+class Oauth2JWTManager(object):
+    _token = ''
+    _token_payload = {}
+    logger = logging.getLogger('oauth2jwt')
+
+
+    @classmethod
+    def set_token(cls, token):
+        cls._token = token
+        cls.set_token_payload()
+        JwtManager.oauth2_token = True
+
+    @classmethod
+    def set_token_payload(cls, token_payload=None):
+        if token_payload:
+            cls._token_payload = token_payload
+        elif cls._token:
+            cls._token_payload = json.loads(base64.urlsafe_b64decode(cls._token.split(".")[1] + "===="))
+
+        if cls._token_payload:
+            JwtManager.oauth2_token = True
+
+    @classmethod
+    def get_token(cls):
+        return cls._token
+
+    @classmethod
+    def _create_user(cls):
+        user_roles = cls.get_user_roles()
+        user = mgr.ACCESS_CTRL_DB.create_user(cls._token_payload['sub'], None, cls._token_payload['name'], cls._token_payload['email'])
```

Comment:
this class needs mgr but we are not passing it as an argument! .. please, consider having a normal object instead of using the static `@classmethod` to call all the methods unless really needed.

Reply:
Changed it